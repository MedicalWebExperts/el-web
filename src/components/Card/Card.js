import React from 'react';
import { shape, node } from 'prop-types';

const propTypes = {
  styles: shape({}),
  children: node.isRequired,
};

const defaultProps = {
  styles: {},
};

const Card = ({
  styles, children,
}) => (
  <div style={styles}>
    <style jsx>
      {`
        div{
          box-shadow: 0 2px 4px rgba(0,0,0,.2);
          padding: 10px;
          border-radius: 4px;
        }
      `}
    </style>
    {children}
  </div>
);

Card.propTypes = propTypes;
Card.defaultProps = defaultProps;

export default Card;
