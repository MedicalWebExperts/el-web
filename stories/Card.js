import React from 'react';
import { storiesOf } from '@storybook/react';

import { Card, Avatar } from '..';

export default () => {
  storiesOf('Card', module).add('Card Default', () => (
    <Card>
      <Avatar image="https://placehold.it/350x350" alt="Image alt" />
    </Card>
  ));
};
