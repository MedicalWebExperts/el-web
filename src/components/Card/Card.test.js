import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

const defaultProps = {
  children: <h1>Card Test</h1>,
  onPress: '',
};

const customStyles = {
  borderRadius: '20px',
  padding: '0',
};

describe('Component Card Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Card {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer.create(<Card {...defaultProps} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
