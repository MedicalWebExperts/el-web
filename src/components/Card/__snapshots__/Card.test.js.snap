// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Card Snapshot using custom styles 1`] = `
<div
  className="jsx-1324345059"
  style={
    Object {
      "borderRadius": "20px",
      "padding": "0",
    }
  }
>
  <a
    className="jsx-1324345059"
    href=""
  >
    <h1>
      Card Test
    </h1>
  </a>
</div>
`;

exports[`Component Card Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-1324345059"
  style={Object {}}
>
  <a
    className="jsx-1324345059"
    href=""
  >
    <h1>
      Card Test
    </h1>
  </a>
</div>
`;
